pkg/forklift-api/webhooks/validating-webhook/admitters/plan-admitter.go

```diff
@@ -0,0 +1,118 @@
+package admitters
+
+import (
+	"context"
+	v1 "k8s.io/api/storage/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"encoding/json"
+	"fmt"
+	admissionv1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"net/http"
+
+	"github.com/konveyor/forklift-controller/pkg/apis"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
+)
+
+type PlanAdmitter struct {
+}
+
+func (admitter *PlanAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	log.Info("Plan admitter was called")
+	raw := ar.Request.Object.Raw
+
+	plan := &api.Plan{}
+	err := json.Unmarshal(raw, plan)
+	if err != nil {
+		return util.ToAdmissionResponseError(err)
+	}
+
+	if plan.Spec.Warm {
+		log.Info("Warm migration supports all storages, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Info("Couldn't get the cluster configuration", "error", err.Error())
+		return util.ToAdmissionResponseAllow()
```

Comment:
why allowing the request when we're not able to validate it here and below?

Reply:
i think i wrote about it in another comment. overall the case is not common to have `no-provisioner` SC. I think if it's documented the validation is more as "nice to have". If something went wrong with the pod or whatever, don't you think it's better to it for every SC?

Reply:
I see it a bit differently - by adding a validation we can ensure that we won't persist invalid configuration. if there's a good argument for persisting it anyway, like we had with providers (the provider may not be accessible when its secret is being defined in MTV), then that's an argument for keep having code that handles such an invalid configuration, but in this case I don't see such an argument - the provider and storage mapping should exist when defining the plan (btw, did you check what happens when defining a plan with a mapping that doesn't exist already?)

Reply:
i checked only non existing mapping (creating within the plan)

Reply:
good, this means that we first create the mappings in this case